app/controllers/concerns/account_reset_concern.rb

```diff
@@ -0,0 +1,35 @@
+module AccountResetConcern
+  include ActionView::Helpers::DateHelper
+  def account_reset_deletion_period_interval
+    current_time = Time.zone.now
+
+    distance_of_time_in_words(
+      current_time,
+      current_time + account_reset_wait_period_days,
+      true,
+      accumulate_on: reset_accumulation_type,
+    )
+  end
+
+  def account_reset_wait_period_days
+    if supports_fraud_account_reset?
+      IdentityConfig.store.account_reset_fraud_user_wait_period_days.days
+    else
+      IdentityConfig.store.account_reset_wait_period_days.days
+    end
+  end
+
+  def supports_fraud_account_reset?
+    (current_user.fraud_review_pending? ||
+      current_user.fraud_rejection?) &&
+      (IdentityConfig.store.account_reset_fraud_user_wait_period_days.present? )
```

Comment:
Style: Mixing `||` and `&&` and splitting across multiple lines makes this a little difficult to follow. Maybe we can at least group the `||` together and drop unnecessary parantheses?

```suggestion
    (current_user.fraud_review_pending? || current_user.fraud_rejection?) &&
      IdentityConfig.store.account_reset_fraud_user_wait_period_days.present?
```

Alternatively we could also split a method or two to get this back to a single line

```suggestion
    fraud_state? && IdentityConfig.store.account_reset_fraud_user_wait_period_days.present?
  end

  def fraud_state?
    current_user.fraud_review_pending? || current_user.fraud_rejection?
```

Reply:
Performance thought: Since a config check is trivial and checking the user involves a database query, and since we expect this would be disabled for a while, we could take advantage of boolean short-circuiting by reversing the order.

e.g.

```rb
# Before
fraud_state? && IdentityConfig.store.account_reset_fraud_user_wait_period_days.present?

# After
IdentityConfig.store.account_reset_fraud_user_wait_period_days.present? && fraud_state?
```